### PR TITLE
ICM-96 Support for publishing artifacts to central maven.

### PR DESCRIPTION
**Pull request by maximkornilov**
Implemented the following changes:
1. Updated gradle documentation to support publishing to maven.
2. Updated travis configuration to publish artifacts to maven.
3. Added encrypted GPG key for artifact signing.
4. Fixed JavaDoc warnings and errors.

Snapshot version is published every commit to master.
Release version is published when new release is created for project using GitHub UI (commit to master with version tag).

ICM-96 "Use Consul as configuration provider for microservices configuration facade"

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,3 +14,23 @@ after_success:
...
+  - chmod +x .travis/deploy.sh
+  - openssl aes-256-cbc -pass pass:$ENCRYPTION_PASSWORD -in .travis/secring.gpg.enc -out local.secring.gpg -d
+
+deploy:
+  -
+    provider: script
+    script: ".travis/deploy.sh"
+    skip_cleanup: true
+    on:
+      repo: intappx/vertx-guice
```
Comment (VolodyaG) [inline .travis.yml]: vertx-guice? It is correct?

File: gradle/publish.gradle [hunk 2]
```diff
@@ -0,0 +1,44 @@
+apply plugin: 'com.bmuschko.nexus'
+
+modifyPom {
```
Comment (VolodyaG) [inline gradle/publish.gradle]: I have some information like group id and etc in gradle.build file for maven-publish plugin. I believe it should be removed, right?

Comment (maximkornilov) [inline gradle/publish.gradle]: It was removed.

File: .travis.yml (hunk 1 above)
Comment (maximkornilov) [inline .travis.yml]: Yes, this is wrong. I will update.

**Pull request merged**